File: homework_1/2.py
n, i, j = map(int, input().split())

# Кол-во станций до последней и до первой.
i_forward = n - i
i_back = i - 1

j_forward = n - j
j_back = j - 1

first_way = (i_forward + j_back) % n
second_way = (i_back + j_forward) % n

print(first_way if first_way < second_way else second_way)




#stations = [s for s in range(1, n + 1)]

File: yandex/algorithm-training-2/homework_1_b/2.cpp
#include <iostream>

int main() {
	int n, i, j;
	std::cin >> n >> i >> j;
	
	// i_forward = n - i, j_back = j - 1
	int first_way = ((n - i) + (j - 1)) % n;
	// i_back = i - 1, j_forward = n - j
	int second_way = ((i - 1) + (n - j)) % n;

	if (first_way > second_way)
		std::cout << second_way;
	else
		std::cout << first_way;

	return 0;
}